src/components/Onfido/index.native.js

```diff
@@ -65,14 +62,14 @@ class Onfido extends React.Component {
                     return;
                 }
 
-                this.props.onError(errorMessage);
+                onError(errorMessage);
             });
-    }
+    }, [onError, onSuccess, onUserExit, sdkToken, translate]);
```

Comment:
Not sure if there should be dependencies here. To match the componentDidMount, this should be an empty array

Reply:
It would require disabling eslint checks for this line. I dont think any of the deps will unexpectedly change and cause unwanted rerender

Reply:
We are already using that eslint disable in several places, so it's common. I would remove dependencies, to maintain the original behaviour. I'll leave it up to you

Reply:
Well, looking at the dependecies passed to this hook. We have 
* sdkToken - it is passed to start the Onfido, if it changes then restarting the whole process would be an exected behaviour so its all good here
* onError - callback wrapped with useCallback and its only dependency is translate, shouldnt cause any extra rerenders
* onSuccess - callback wrapped with useCallback and its only dependency is applicantID. If this changes then rerender would be expected behaviour too imo
onUserExit - thats just Navigate.goBack wrapped with useCallback and no deps

I'd leave it the way it is. I think we should only disable eslint/typescript only when absolutely necessary

Reply:
we should really avoid disabling eslint checks especially for deps, rules of hooks are there for a reason ;) it's also quite concerning that it's a common practice in the codebase

Reply:
I don't think it makes sense for these parameters to update, onfido should only be initialized once. Check `BaseOnfidoWeb`, it uses a `useEffect` with no dependencies, and their justification is that '// Onfido should be initialized only once on mount'.

Reply:
okay, ill change it so it matches the web version